### PR TITLE
Fix typos and invalid link in the yorkie design document

### PR DESCRIPTION
**Pull request by g2hhh2ee**
**What this PR does / why we need it**:
Fix typos and invalid link in yorkie design document

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: design/data-structure.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ JSON-like data strucutres are used when editing JSON-like documents.
 
 - `Primitive`: represents primitive data like `string`, `number`, `boolean`, `null`, etc.
 - `Object`: represents [object type](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) of JavaScript. Just like JavaScript, you can use `Object` as [hash table](https://en.wikipedia.org/wiki/Hash_table).
-- `Array`: represents [array type](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) of JavaScript. You can also use `Array` as [list](https://en.wikipedia.org/wiki/List_(abstract_data_type)).
+- `Array`: represents [array type](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) of JavaScript. You can also use `Array` as [list](<https://en.wikipedia.org/wiki/List_(abstract_data_type)>).
```
Comment (krapie) [inline design/data-structure.md:47]: What are these `<`, `>` for?

**Review by krapie — COMMENTED** [review]
Thank you for your contribution 😄

I left a small comment below.

File: design/data-structure.md (hunk 1 above)
Comment (se030) [reply design/data-structure.md]: For some reasons my prettier added these. Removed them in d8427de4584e3faaeb8bd5531de92dadacf99eba.

**Review by krapie — APPROVED** [review]
LGTM 👍🏼

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**